### ISSUE TITLE
`install` doesn't hard link to the store on XFS

### ISSUE THREAD

**Issue by h3xx** [pnpm/pnpm#6066]
The install process doesn't create any hard links on XFS volumes, though the install process still succeeds.

Interestingly, running `pnpm i --package-import-method hardlink` causes hard links to be created even on XFS volumes.

Therefore, I'm thinking the issue lies in the test for the ability to hardlink with `--package-import-method auto` (the default).

**This issue is specific to XFS volumes**, but may affect other filesystems. I was not able to reproduce this issue using an EXT4 volume in place of an XFS volume.

### Code to reproduce the issue:

Make a temporary loop device with XFS:
```sh
# (as root)
truncate -s 1G data.raw
mkfs.xfs data.raw
mkdir -p data
mount -o loop data.raw data
cd data
git clone https://github.com/pnpm/pnpm
cd pnpm
pnpm i
```

Then test:
```sh
find node_modules -type f ! -links 1
```
If no lines are printed, hard linking failed.

## Control test using EXT4

<details>
<summary>I tried this and it succeeded using an EXT4 volume:</summary>

Make a temporary loop device with EXT4:
```sh
# (as root)
truncate -s 1G data.raw
mkfs.ext4 data.raw
mkdir -p data
mount -o loop data.raw data
cd data
git clone https://github.com/pnpm/pnpm
cd pnpm
pnpm i
```

Then test:
```sh
find node_modules -type f ! -links 1
```
If no lines are printed, the issue also affects EXT4 and I was wrong about it just affecting XFS.
</details>

### Expected behavior:

Some hard links should be created in `node_modules`.

### Actual behavior:

No hard links are created in `node_modules`.

### Additional information:

 - `node -v` prints: `v19.6.0`
 - Windows, macOS, or Linux?: Linux

**Comment by octylFractal** [comment]
I can report a similar issue on Linux 6.2.2-zen1-1-zen with BTRFS. `node -v` is `v17.3.0`. `pnpm i --package-import-method hardlink` does workaround this issue.

**Comment by octylFractal** [comment]
After investigating, I don't think this is a bug, but a feature: BTRFS and XFS support `FICLONE`, so when `auto` [tests for FICLONE](https://github.com/pnpm/pnpm/blob/main/fs/indexed-pkg-importer/src/index.ts#L56) it succeeds, and it uses that instead (can be seen in the log as "Packages are **cloned** from the content-addressable store to the virtual store." (emphasis mine)).

I was able to prove that it was successfully being FICLONE'd by checking `filefrag` (the physical offsets are the same):
```sh
$ filefrag -e node_modules/.pnpm/is-number@6.0.0/node_modules/is-number/README.md ~/.local/share/pnpm/store/v3/files/58/01e592e94aa4076c179180956f33f8262fa7157a06bfe636a6854bd3b9b2ddf44780ffa1405642ad8ece2ed6514e19a28178cbecdcf5608a3be710bbf3c0d4
Filesystem type is: 9123683e
File size of node_modules/.pnpm/is-number@6.0.0/node_modules/is-number/README.md is 5766 (2 blocks of 4096 bytes)
 ext:     logical_offset:        physical_offset: length:   expected: flags:
   0:        0..       1:    5071437..   5071438:      2:             last,encoded,shared,eof
node_modules/.pnpm/is-number@6.0.0/node_modules/is-number/README.md: 1 extent found
File size of /home/octy/.local/share/pnpm/store/v3/files/58/01e592e94aa4076c179180956f33f8262fa7157a06bfe636a6854bd3b9b2ddf44780ffa1405642ad8ece2ed6514e19a28178cbecdcf5608a3be710bbf3c0d4 is 5766 (2 blocks of 4096 bytes)
 ext:     logical_offset:        physical_offset: length:   expected: flags:
   0:        0..       1:    5071437..   5071438:      2:             last,encoded,shared,eof
/home/octy/.local/share/pnpm/store/v3/files/58/01e592e94aa4076c179180956f33f8262fa7157a06bfe636a6854bd3b9b2ddf44780ffa1405642ad8ece2ed6514e19a28178cbecdcf5608a3be710bbf3c0d4: 1 extent found
```

**Comment by zkochan** [comment]
Yes, this works as expected. By default, pnpm uses [package-import-method=auto](https://pnpm.io/npmrc#package-import-method), which will create a reference link, if possible. If a reference link is not supported by the filesystem, then a hard link will be created. In the worst case scenario, regular copies are created.
